roles/logging/tasks/main.yml

```diff
@@ -4,29 +4,32 @@
   with_fileglob: ../templates/etc/rsyslog.d/*
   notify: restart rsyslog
 
-- name: stop logstash-forwarder service
-  service: name=logstash-forwarder state=stopped enabled=yes
-  register: lsfstop
-  failed_when: lsfstop|failed and ("service not found" not in lsfstop.msg)
-
 - name: create logstash user
   user: name=logstash comment="Logstash Service User" shell=/sbin/nologin system=yes home=/var/lib/logstash
 
 - name: install logstash-forwarder
   apt: pkg=logstash-forwarder
-  when: ansible_distribution_version == "12.04"
+  when: logging.install_method == 'apt'
+
+- name: stop logstash-forwarder service
+  service: name=logstash-forwarder state=stopped enabled=yes
+  register: lsfstop
+  failed_when: lsfstop|failed and ("service not found" not in lsfstop.msg)
```

Comment:
Unrelated to moving this around, I don't like that we do this unconditionally every run of Ansible.